### PR TITLE
Purge data for auditor and dedupe tables

### PR DESCRIPTION
**Pull request by javierscope**

### CODE REVIEW

**Review by ravi-zadpide — CHANGES_REQUESTED** [review]

File: src/main/java/com/scoperetail/fusion/auditor/eraser/CommandImpl.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  private final MessageLogKeyRepository messageLogKeyRepository;
+
+  public CommandImpl(MessageLogRepository messageLogRepository, MessageLogKeyRepository messageLogKeyRepository) {
+    this.messageLogRepository = messageLogRepository;
+    this.messageLogKeyRepository = messageLogKeyRepository;
+  }
+
+  private static final String MESSAGE_LOG_TABLE_NAME = "message_log";
+  private static final String MESSAGE_LOG_KEY_TABLE_NAME = "message_log_key";
+  private static final String LOG_FILES = "auditor_log_file";
```
[the remark this answers is not in the document]
Comment (javierscope) [reply src/main/java/com/scoperetail/fusion/auditor/eraser/CommandImpl.java:32]: Done

File: src/main/java/com/scoperetail/fusion/auditor/eraser/CommandImpl.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+  @Value("${retentionDuration}")
+  private Integer retentionDuration;
+
+  @Value("${logsDirectory}")
+  private String logsDirectory;
```
[the remark this answers is not in the document]
Comment (javierscope) [reply src/main/java/com/scoperetail/fusion/auditor/eraser/CommandImpl.java:37]: done

File: src/main/java/com/scoperetail/fusion/auditor/config/SpringAuditorConfig.java [hunk 3]
```diff
@@ -1,14 +1,20 @@
...
 import lombok.AllArgsConstructor;
+import org.springframework.scheduling.annotation.EnableScheduling;
 
 @Configuration
 @EnableConfigurationProperties
 @AllArgsConstructor
-@Import({FusionCoreConfig.class})
+@Import({FusionCoreConfig.class, ConfigProperties.class, SchedulerConfig.class})
+@EnableScheduling
+@ComponentScan(basePackages = {"com.scoperetail"})
```
[the remark this answers is not in the document]
Comment (javierscope) [reply src/main/java/com/scoperetail/fusion/auditor/config/SpringAuditorConfig.java]: Solved

File: src/main/java/com/scoperetail/fusion/auditor/config/SpringAuditorConfig.java [hunk 4]
```diff
@@ -1,14 +1,20 @@
...
 import org.springframework.context.annotation.Import;
 import com.scoperetail.fusion.core.FusionCoreConfig;
 import lombok.AllArgsConstructor;
+import org.springframework.scheduling.annotation.EnableScheduling;
 
 @Configuration
 @EnableConfigurationProperties
 @AllArgsConstructor
-@Import({FusionCoreConfig.class})
+@Import({FusionCoreConfig.class, ConfigProperties.class, SchedulerConfig.class})
```
[the remark this answers is not in the document]
Comment (javierscope) [reply src/main/java/com/scoperetail/fusion/auditor/config/SpringAuditorConfig.java]: Solved

File: src/main/java/com/scoperetail/fusion/auditor/eraser/CommandImpl.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (javierscope) [reply src/main/java/com/scoperetail/fusion/auditor/eraser/CommandImpl.java]: Solved

File: src/main/java/com/scoperetail/fusion/auditor/eraser/CommandImpl.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (javierscope) [reply src/main/java/com/scoperetail/fusion/auditor/eraser/CommandImpl.java]: Solved

File: src/main/resources/application-local.yml [hunk 5]
```diff
@@ -29,4 +29,17 @@ spring:
...
+  duration: 60
+  flushStatsPerCycle: true
+  flushStatsPerOperation: false
+  frequency: '0 27 10 * * ?'
+  isEnabled: true
+  shedlock:
+    lockAtLeastFor: 1m
+    lockAtMostFor: 10m
+retentionDuration: 15
+logsDirectory: '/tmp/fusion/auditor/'
```
[the remark this answers is not in the document]
Comment (javierscope) [reply src/main/resources/application-local.yml]: Solved

File: src/main/resources/application-walmart_prod.yml [hunk 6]
```diff
@@ -29,4 +29,17 @@ spring:
...
+  duration: 60
+  flushStatsPerCycle: true
+  flushStatsPerOperation: false
+  frequency: '0 0 22 * * ?'
+  isEnabled: true
+  shedlock:
+    lockAtLeastFor: 1m
+    lockAtMostFor: 10m
+retentionDuration: 15
+logsDirectory: '/opt/fusion/target/fusion/auditor/'
```
[the remark this answers is not in the document]
Comment (javierscope) [reply src/main/resources/application-walmart_prod.yml]: Solved

File: src/main/resources/application-walmart_qa.yml [hunk 7]
```diff
@@ -29,4 +29,17 @@ spring:
...
+  duration: 60
+  flushStatsPerCycle: true
+  flushStatsPerOperation: false
+  frequency: '0 0 22 * * ?'
+  isEnabled: true
+  shedlock:
+    lockAtLeastFor: 1m
+    lockAtMostFor: 10m
+retentionDuration: 15
+logsDirectory: '/opt/fusion/target/fusion/auditor/'
```
[the remark this answers is not in the document]
Comment (javierscope) [reply src/main/resources/application-walmart_qa.yml]: Solved

**Pull request merged**